pkg/controllers/schema/schemas.go

```diff
@@ -88,7 +132,28 @@ func (h *handler) queueRefresh() {
 
 	go func() {
 		time.Sleep(500 * time.Millisecond)
-		if err := h.refreshAll(h.ctx); err != nil {
+		var err error
+		var changedIDs map[k8sapimachineryschema.GroupVersionKind]bool
+		var deletedCRDs map[k8sapimachineryschema.GroupVersionKind]bool
+		var createdCRDs map[k8sapimachineryschema.GroupVersionKind]bool
+		h.Lock()
+		if len(h.createdCRDs) > 0 {
+			createdCRDs = h.createdCRDs
+			h.createdCRDs = make(map[k8sapimachineryschema.GroupVersionKind]bool)
+		}
+		if len(h.deletedCRDs) > 0 {
+			deletedCRDs = h.deletedCRDs
+			h.deletedCRDs = make(map[k8sapimachineryschema.GroupVersionKind]bool)
+		}
+		if len(h.changedIDs) > 0 {
+			changedIDs = h.changedIDs
+			h.changedIDs = make(map[k8sapimachineryschema.GroupVersionKind]bool)
+		}
+		h.Unlock()
+		if len(changedIDs) > 0 || len(deletedCRDs) > 0 || len(createdCRDs) > 0 {
+			err = h.refreshAll(h.ctx, changedIDs, len(deletedCRDs) > 0 || len(createdCRDs) > 0)
```

Comment:
I think it works, but it does feel/look awkward with the compound conditional used as the parameter.  Maybe split that out for clarity?

Reply:
Fixed